taichi/ir/control_flow_graph.h

```diff
@@ -15,17 +19,27 @@ class CFGNode {
   CFGNode *prev_node_in_same_block;
   CFGNode *next_node_in_same_block;
 
+  // Edges in the graph
   std::vector<CFGNode *> prev, next;
 
+  // Reaching definition analysis
+  std::unordered_set<Stmt *> reach_gen, reach_kill, reach_in, reach_out;
```

Comment:
We should probably point people to something like https://en.wikipedia.org/wiki/Reaching_definition